synapse/config/experimental.py

```diff
@@ -27,3 +27,9 @@ def read_config(self, config: JsonDict, **kwargs):
 
         # MSC2858 (multiple SSO identity providers)
         self.msc2858_enabled = experimental.get("msc2858_enabled", False)  # type: bool
+
+        # MSC2946 (space summaries)
+        self.msc2946_enabled = experimental.get("msc2946_enabled", False)  # type: bool
+
+        # MSC1772 (spaces)
+        self.msc1772_enabled = experimental.get("msc1772_enabled", False)  # type: bool
```

Comment:
I see in the code where these are used, but it seems like it doesn't really make sense to enable one without the other, do we really need two flags?

Reply:
Good question. I'm trying to look forward to a time when we start dropping support for these unstable identifiers, and wanted to leave the path open for them to land separately

I kindof agree though. I'll combine them into one option.